### PR TITLE
Process import batch

### PR DESCRIPTION
**Pull request by josephbridgwaterrowe**

### CODE REVIEW

**Comment by josephbridgwaterrowe** [comment]
@michaelachrisco Please review and merge if all seems in order.

File: app/interactors/import/create_batch.rb [hunk 1]
```diff
@@ -1,40 +1,37 @@
...
-  before :validate
+class Import::CreateBatch
+  include Interactor::Creator
 
   def call
-    # do save in before block
+    model.whodunnit(context.user) { model.save! }
 
-    # queue file processing
+    # TODO: Queue processing
```
Comment (michaelachrisco) [inline app/interactors/import/create_batch.rb]: Is this TODO needed for the master branch?
Might be better to make it an issue in Github if it is going to be a new feature.

File: app/interactors/import/create_batch.rb [hunk 2]
```diff
@@ -1,40 +1,37 @@
...
-    end
+    context.message = 'Batch successfully created'
   end
 
-  def create_batch
-    context.batch = Import::Batch.new(context.to_h.except(:upload_file_path, :upload_file))
-    context.batch.status = :new
+  def after_build
+    model.status = :new
     # TODO: Refactor this
```
Comment (michaelachrisco) [inline app/interactors/import/create_batch.rb]: Another TODO.

File: spec/features/user_creates_new_quantity_translation_spec.rb [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
-  
+
   scenario 'visit new page' do
     visit new_quantity_translation_path
 
-    expect(page).to have_content(/new price translation/i)
+    expect(page).to have_content(/new quantity translation/i)
   end
 
   # TODO: Add a javascript driver so we can test the Entity typeahead
```
Comment (michaelachrisco) [inline spec/features/user_creates_new_quantity_translation_spec.rb]: Another TODO. What is your opinion of TODO's in testing?

File: spec/features/user_creates_new_unit_of_measure_translation_spec.rb [hunk 4]
```diff
@@ -12,17 +12,22 @@
   end
 
   # TODO: Add a javascript driver so we can test the Entity typeahead
```
Comment (michaelachrisco) [inline spec/features/user_creates_new_unit_of_measure_translation_spec.rb]: Another TODO.

File: spec/features/user_deletes_import_batch_spec.rb [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  given(:user) { create(:user, :confirmed) }
+  
+  scenario 'batch is deleted' do
+    visit entities_path
+
+    # TODO: Click the first delete button
+
+    # TODO: Check for a delete confirmation flash
+
+    # TODO: Check that the batch has been deleted
```
Comment (michaelachrisco) [inline spec/features/user_deletes_import_batch_spec.rb]: 3 TODO's.

File: app/interactors/import/create_batch.rb (hunk 1 above)
Comment (josephbridgwaterrowe) [inline app/interactors/import/create_batch.rb]: Yes, good point... I will do that.

File: app/interactors/import/create_batch.rb (hunk 2 above)
Comment (josephbridgwaterrowe) [inline app/interactors/import/create_batch.rb]: I'm going to cheat and change this to a REVIEW: :wink:

File: spec/interactors/import/create_batch_spec.rb [hunk 6]
```diff
@@ -25,15 +30,31 @@
...
+      ActionDispatch::Http::UploadedFile.new(
+        :type => 'text/csv',
+        :tempfile => file,
+        :filename => File.basename(file),
+        :head => [
+          'Content-Disposition: form-data',
+          'name="import_batch_input[upload_file]"',
+          "filename=\"#{File.basename(file.path)}\"",
+          'Content-Type: text/csv\r\n'].join(';'))
+    end
```
Comment (michaelachrisco) [inline spec/interactors/import/create_batch_spec.rb]: Im thinking this should read: 
```Ruby
    let(:upload_file) do
      ActionDispatch::Http::UploadedFile.new(
        :type => 'text/csv',
        :tempfile => file,
        :filename => File.basename(file),
        :head => [
          'Content-Disposition: form-data',
          'name="import_batch_input[upload_file]"',
          "filename=\"#{File.basename(file.path)}\"",
          'Content-Type: text/csv\r\n']
        .join(';'))
      end
```
Or head should be its own let statement.

Comment (josephbridgwaterrowe) [inline spec/interactors/import/create_batch_spec.rb]: I now have this...

```ruby
    let(:upload_file) do
      ActionDispatch::Http::UploadedFile.new(
        :type => 'text/csv',
        :tempfile => file,
        :filename => File.basename(file),
        :head => [
          'Content-Disposition: form-data',
          'name="import_batch_input[upload_file]"',
          "filename=\"#{File.basename(file.path)}\"",
          'Content-Type: text/csv\r\n'
        ].join(';')
      )
    end
```

Agree?

Comment (michaelachrisco) [inline spec/interactors/import/create_batch_spec.rb]: :+1:

**Comment by josephbridgwaterrowe** [comment]
@michaelachrisco, I think this is good to merge if you ignore the hound issues.

**Pull request merged**